### PR TITLE
Ports get_time_space_phys_variables subroutine

### PR DESCRIPTION
**Pull request by singhbalwinder**
[BFB]

### CODE REVIEW

File: components/scream/cmake/machine-files/generic-desktop-debug.cmake [hunk 1]
```diff
@@ -1,5 +0,0 @@
-set(KOKKOS_ENABLE_DEBUG TRUE CACHE BOOL "")
-set(KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION FALSE CACHE BOOL "")
-set(KOKKOS_ENABLE_OPENMP TRUE CACHE BOOL "")
```
Comment (jgfouca) [inline components/scream/cmake/machine-files/generic-desktop-debug.cmake]: Why was this deleted?

File: components/scream/src/physics/p3/p3_functions_f90.cpp [hunk 2]
```diff
@@ -994,6 +1003,56 @@ void get_rain_dsd2_f(Real qr_, Real* nr_, Real* mu_r_, Real* lamr_, Real* cdistr
...
+void get_time_space_phys_variables_f(Real t_, Real pres_, Real rho_, Real xxlv_, Real xxls_, Real qvs_, Real qvi_,
+				     Real* mu_, Real* dv_, Real* sc_, Real* dqsdt_, Real* dqsidt_, Real* ab_,
+				     Real* abi_, Real* kap_, Real* eii_)
+{
+  using P3F = Functions<Real, DefaultDevice>;
+
+  typename P3F::view_1d<Real> t_d("t_h", 9);
+  auto t_h = Kokkos::create_mirror_view(t_d);
+
+  Real local_mu     = *mu_;
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_f90.cpp]: I don't think any of these local_* are needed.

File: components/scream/src/physics/p3/p3_functions_f90.hpp [hunk 3]
```diff
@@ -739,6 +739,26 @@ void compute_rain_fall_velocity_f(Real qr_incld, Real rcldm, Real rhofacr,
...
+  //Outs
+  Real mu, dv, sc, dqsdt, dqsidt, ab, abi, kap, eii;
+};
+
+void get_time_space_phys_variables(GetTimeSpacePhysVarsData& d);
+
+extern "C"{
+
+void get_time_space_phys_variables_f(Real t, Real pres, Real rho, Real xxlv, Real xxls, Real qvs, Real qvi,
+Real* mu, Real* dv, Real* sc, Real* dqsdt, Real* dqsidt, Real* ab, Real* abi, Real* kap, Real* eii);
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_f90.hpp]: Need indent here.

File: components/scream/src/physics/p3/p3_functions_get_time_space_phys_variables.cpp [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace scream {
+namespace p3 {
+
+/*
+ * Explicit instantiation for doing p3 get_time_space_phys_variables function on Reals using the
+ * default device.
+ */
+
+  template struct Functions<Real,DefaultDevice>;
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_get_time_space_phys_variables.cpp]: No indent here.

File: components/scream/src/physics/p3/p3_functions_get_time_space_phys_variables_impl.hpp [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::get_time_space_phys_variables(const Spack& t, const Spack& pres, const Spack& rho, const Spack& xxlv, const Spack& xxls,
+				const Spack& qvs, const Spack& qvi, Spack& mu, Spack& dv, Spack& sc, Spack& dqsdt,
+				Spack& dqsidt, Spack& ab, Spack& abi, Spack& kap, Spack& eii)
+{
+  //time/space varying physical variables
+  mu = sp(1.496e-6) * pow(t,sp(1.5))/(t+sp(120));
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_get_time_space_phys_variables_impl.hpp]: Just do `120`, not `sp(120)`

File: components/scream/cmake/machine-files/generic-desktop-debug.cmake (hunk 1 above)
Comment (singhbalwinder) [reply components/scream/cmake/machine-files/generic-desktop-debug.cmake]: It got deleted during a rebase. Thanks for catching that. I have reverted this change.

File: components/scream/src/physics/p3/p3_functions_f90.cpp (hunk 2 above)
Comment (singhbalwinder) [reply components/scream/src/physics/p3/p3_functions_f90.cpp]: I am a little confused here. These are intent outs. Don't we need them in the following:
```
typename P3F::Spack mu(local_mu), dv(local_dv), sc(local_sc), dqsdt(local_dqsdt),
        dqsidt(local_dqsidt), ab(local_ab), abi(local_abi), kap(local_kap), eii(local_eii);
```

Comment (jgfouca) [reply components/scream/src/physics/p3/p3_functions_f90.cpp]: All these values are intent(out), so there's no need to get there values as inputs. You can initialize the corresponding Spacks with their default constructor.

Comment (singhbalwinder) [reply components/scream/src/physics/p3/p3_functions_f90.cpp]: Thanks! I have fixed it now.

**Comment by jgfouca** [comment]
@singhbalwinder , this should work now.

**Comment by jgfouca** [comment]
Tested white by hand and it worked, merging.
